### DISCUSSION TITLE
C# / .NET Native Interop with Tauri

### DISCUSSION THREAD

**Discussion by velocitysystems** [tauri-apps/tauri#11932] [General]
Hello! 👋 I’m currently evaluating Tauri to replace React Native for a cross platform iOS, Android and Windows app.

Tauri 2.x is a great leap forward in being able to write plugins in Swift/Kotlin, however for Windows you are limited to interacting with Win32 APIs via Rust. What is the feasibility of adding interop support for writing plugins in C# with .NET 8/9 or .NET Native? There are many new AOT features available and 100,000s of NuGet packages available in the ecosystem.

Reading through the documentation it appears it would require:
- Interop support
- Tooling support
- Debugging support

[Daniel Thompson-Yvetot's webinar](https://www.youtube.com/watch%3Fv%3DpT-UuXrsviE&ved=2ahUKEwj4scTPj52KAxWhWEEAHaR0IiEQtwJ6BAgQEAI&usg=AOvVaw0S9i8_cSWgata3uJ-ysKvW) mentioned the prospect of additional language support in Tauri 3.x. Is this an effort that the community can also be involved in?

**Comment by FabianLars** [comment]
> Is this an effort that the community can also be involved in?

as it stands right now, it's more like we _need_ as much community involvement as possible. we currently don't even have time to steward the general approach tbh. The general tracking issue is this one btw https://github.com/tauri-apps/tauri/issues/4670

**Comment by velocitysystems** [comment]
Thanks @FabianLars. Will give this some further thought. Should a separate tracking issue be created for each additional proposed language? That would make it easier to track and discuss language-specific requirements.

**Reply by FabianLars** [reply]
i'd prefer not to, at least not yet.

**Comment by snspinn** [comment]
What about the interface when running as a [sidecar](https://tauri.app/develop/sidecar/)? Seems to be text/stdout only but can this be "streaming" or does this work with one-shot binaries?
